tools/src/test/java/org/apache/kafka/tools/consumer/group/DescribeConsumerGroupTest.java

```diff
@@ -0,0 +1,855 @@
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import scala.Function0;
+import scala.Function1;
+import scala.Option;
+import scala.collection.Seq;
+import scala.runtime.BoxedUnit;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.apache.kafka.test.TestUtils.DEFAULT_MAX_WAIT_MS;
+import static org.apache.kafka.test.TestUtils.RANDOM;
+import static org.apache.kafka.tools.ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES;
+import static org.apache.kafka.tools.ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_NAME;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class DescribeConsumerGroupTest extends ConsumerGroupCommandTest {
+    private static final String[][] DESCRIBE_TYPE_OFFSETS = new String[][]{new String[]{""}, new String[]{"--offsets"}};
+    private static final String[][] DESCRIBE_TYPE_MEMBERS = new String[][]{new String[]{"--members"}, new String[]{"--members", "--verbose"}};
+    private static final String[][] DESCRIBE_TYPE_STATE = new String[][]{new String[]{"--state"}};
+    private static final String[][] DESCRIBE_TYPES;
+
+    static {
+        List<String[]> describeTypes = new ArrayList<>();
+
+        describeTypes.addAll(Arrays.asList(DESCRIBE_TYPE_OFFSETS));
+        describeTypes.addAll(Arrays.asList(DESCRIBE_TYPE_MEMBERS));
+        describeTypes.addAll(Arrays.asList(DESCRIBE_TYPE_STATE));
+
+        DESCRIBE_TYPES = describeTypes.toArray(new String[0][0]);
+    }
+
+    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
+    @MethodSource({"getTestQuorumAndGroupProtocolParametersAll"})
+    public void testDescribeNonExistingGroup(String quorum, String groupProtocol) {
+        createOffsetsTopic(listenerName(), new Properties());
+        String missingGroup = "missing.group";
+
+        for (String[] describeType : DESCRIBE_TYPES) {
+            // note the group to be queried is a different (non-existing) group
+            List<String> cgcArgs = new ArrayList<>(Arrays.asList("--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--group", missingGroup));
+            cgcArgs.addAll(Arrays.asList(describeType));
+            ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs.toArray(new String[0]));
+
+            String output = kafka.utils.TestUtils.grabConsoleOutput(describeGroups(service));
```

Comment:
Can `kafka.utils.TestUtils.grabConsoleOutput` be replaced by `ToolsTestUtils#captureStandardOut`?

Reply:
Not for now.

Scala caches `System.out` value inside `Console#outVar` and when we change `System.out` value from java code we can't capture scala code output.

But, in big PR (which moves command itself) I use `ToolsTestUtils#captureStandartOut` instead of scala version.